cmd/clientapi/main.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"flag"
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/notary/client_api/api"
+	"github.com/docker/notary/cmd/clientapi/setup"
+	"github.com/docker/notary/utils"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+var (
+	configPath string
+)
+
+func init() {
+	flag.StringVar(
+		&configPath,
+		"config",
+		"config.toml",
+		"path to configuration file; supported formats are JSON, YAML, and TOML",
+	)
+	flag.Parse()
+}
+
+func main() {
+	vc, err := setup.ViperConfig(configPath)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+
+	logrus.SetLevel(logrus.DebugLevel)
+
+	// FIXME: refactor with notary-signer/config.go:getAddrAndTLSConfig()
+	// and setupGRPCServer()
+	grpcAddr, tlsConfig, err := setup.GetAddrAndTLSConfig(vc)
+	if err != nil {
+		logrus.Fatal("unable to set up TLS: %s", err.Error())
```

Comment:
I think we should allow running without TLS/Auth. 

I know this isn't the main use-case being considered, but I was excited about the idea of running as a daemon locally. Your notary client would connect to your notary client daemon running locally, which you don't have ACLs on because it just uses whatever keys you gave it through normal notary operations.

This is cool because it allows other applications to interact with notary on the user's behalf (obviously they'd still need to unlock access for e.g. local keychain and verify the operations). I think it would be a powerful way for notary to integrate with other tools.

You could require TLS on the loopback but the Authorization part seems completely unnecessary - it doesn't make sense to auth with an external service to access the keys on my machine.